backend/Controller/Entities/Coords.cs

```diff
@@ -12,14 +14,35 @@ public Coords(decimal longitude, decimal latitude)
             Latitude = latitude;
         }
 
-        private decimal ManhattanDistance(Coords others)
-        {
-            return Math.Abs(Longitude - others.Longitude) + Math.Abs(Latitude - others.Latitude);
+        private double HaversineDistanceKM(Coords others)
+        {
+            double latDistance = ToRadians(Latitude - others.Latitude);
+            double lngDistance = ToRadians(Longitude - others.Longitude);
+
+            double a =
+                Math.Sin((double)latDistance / 2.0) * Math.Sin((double)latDistance / 2.0)
+              + Math.Cos(ToRadians(Latitude)) * Math.Cos(ToRadians(others.Latitude))
+              * Math.Sin(lngDistance / 2) * Math.Sin(lngDistance / 2);
+
+            double c = 2 * Math.Atan2(Math.Sqrt(a), Math.Sqrt(1 - a));
+
+            return AVERAGE_RADIUS_OF_EARTH_KM * c;
+        }
```

Comment:
Maybe leave a quick comment why this is used, because it's not clear from just looking at the code. A link to some explanations could be nice too.